tests/src/Functional/UserSubscriptionsPageTestBase.php

```diff
@@ -271,6 +274,50 @@ protected function doTestUserPreferences(UserInterface $user, callable $fn_get_p
     $this->assertEquals('it', $user->getPreferredLangcode());
   }
 
+  /**
+   * Tests the rendering of the introduction text.
+   *
+   * @param \Drupal\user\UserInterface $user
+   *   The user for the test.
+   * @param callable $fn_get_path
+   *   Callable that returns the path to visit for the test. It also needs to
+   *   take care of logging the user, or requesting a token in order to allow
+   *   visiting the path.
+   */
+  protected function doTestFormPreface(UserInterface $user, callable $fn_get_path): void {
+    $path = $fn_get_path($user);
+    $this->drupalGet($path);
+    $assert_session = $this->assertSession();
+    $subscriptions_config = \Drupal::configFactory()->getEditable(SettingsForm::CONFIG_NAME);
+
+    // Test that text isn't present by default.
+    $this->drupalGet($path);
+    $assert_session->pageTextNotContains('Configurable text 1.');
+
+    // Test that the text is displayed after setting configuration.
+    $subscriptions_config
+      ->set('introduction_text', [
+        'value' => 'Configurable text 1.',
+        'format' => 'plain_text',
+      ])->save();
+    $this->drupalGet($path);
+    $assert_session->pageTextContains('Configurable text 1.');
+
+    // Test that the text with a link is displayed after updating configuration.
+    FilterFormat::create([
+      'format' => 'full_html',
+      'name' => 'Full HTML',
+    ])->save();
+    $subscriptions_config
+      ->set('introduction_text', [
+        'value' => 'Configurable text 2 <a href="' . Url::fromRoute('<front>')->toString() . '">Terms and conditions</a>.',
+        'format' => 'full_html',
+      ])->save();
+    $this->drupalGet($path);
+    $assert_session->pageTextContains('Configurable text 2 Terms and conditions.');
+    $assert_session->linkExists('Terms and conditions');
```

Comment:
You need to find the link and assert its href.

Reply:
ok